### PR TITLE
🌐 [i18n-KO] Translated `auto.md` to Korean

### PR DESCRIPTION
**Pull request by boyunJang**
# What does this PR do?

Translated the `auto.md` file of the documentation to Korean.
Thank you in advance for your review. 

Part of https://github.com/huggingface/transformers/issues/20179

## Before reviewing
- [x] Check for missing / redundant translations (번역 누락/중복 검사)
- [x] Grammar Check (맞춤법 검사)
- [x] Review or Add new terms to glossary (용어 확인 및 추가)
- [x] Check Inline TOC (e.g. `[[lowercased-header]]`)
- [ ] Check live-preview for gotchas (live-preview로 정상작동 확인)

## Who can review? (Initial)

May you please review this PR?
 
@junejae, @Jwaminju, @010kim, @4N3MONE, @boyunJang
 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review? (Final)

@stevhliu May you please review this PR?

### CODE REVIEW

File: docs/source/ko/model_doc/auto.md [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+specific language governing permissions and limitations under the License.
+
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+
+# 자동 클래스[[auto-classes]]
+
+많은 경우, 사용하려는 아키텍처는 `from_pretrained()` 메소드에서 제공하는 사전 학습된 모델의 이름이나 경로로부터 유추할 수 있습니다. AutoClasses는 이 작업을 위해 존재하며, 사전 학습된 weight/config/vocabulary에 대한 이름/경로를 제공하면 자동으로 관련 모델을 가져오도록 도와줍니다.
```
Comment (Jwaminju) [inline docs/source/ko/model_doc/auto.md:19]:
```suggestion
많은 경우, 사용하려는 아키텍처는 `from_pretrained()` 메소드에서 제공하는 사전훈련된 모델의 이름이나 경로로부터 유추할 수 있습니다. AutoClasses는 이 작업을 위해 존재하며, 사전 학습된 weight/config/vocabulary에 대한 이름/경로를 제공하면 자동으로 관련 모델을 가져오도록 도와줍니다.
```
glossary 반영해보았습니다!

File: docs/source/ko/model_doc/auto.md [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+
+# 자동 클래스[[auto-classes]]
```
Comment (Jwaminju) [inline docs/source/ko/model_doc/auto.md:17]:
```suggestion
# Auto 클래스[[auto-classes]]
```
auto class가 사람이름 같이 고유명사인 것 같아서 제안해보아요!

File: docs/source/ko/model_doc/auto.md [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+각 자동 클래스는 사용자의 커스텀 클래스로 확장될 수 있는 메소드를 가지고 있습니다. 예를 들어, `NewModel`이라는 커스텀 모델 클래스를 정의했다면, `NewModelConfig`를 준비한 후 다음과 같이 자동 클래스에 추가할 수 있습니다:
+
+```python
+from transformers import AutoConfig, AutoModel
+
+AutoConfig.register("new-model", NewModelConfig)
+AutoModel.register(NewModelConfig, NewModel)
+```
+
+그 후에는 평소처럼 자동 클래스를 사용할 수 있게 됩니다!
```
Comment (Jwaminju) [inline docs/source/ko/model_doc/auto.md:43]:
```suggestion
이후에는 평소처럼 자동 클래스를 사용할 수 있게 됩니다!
```
자연스럽게 바꾸어보았습니다!

File: docs/source/ko/model_doc/auto.md [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+
+[[autodoc]] TFAutoModel
+
+### FlaxAutoModel[[transformers.FlaxAutoModel]]
+
+[[autodoc]] FlaxAutoModel
+
+## 일반적인 사전 학습 클래스[[generic-pretraining-classes]]
+
+다음 자동 클래스들은 사전 학습 헤드가 포함된 모델을 인스턴스화하는 데 사용할 수 있습니다.
```
Comment (Jwaminju) [inline docs/source/ko/model_doc/auto.md:91]:
```suggestion
다음 자동 클래스들은 사전 훈련 헤드가 포함된 모델을 인스턴스화하는 데 사용할 수 있습니다.
```
pretrained model이 glossary에서 사전훈련으로 번역하고 있어서, 통일해보았습니다.

File: docs/source/ko/model_doc/auto.md (hunk 1 above)
Comment (4N3MONE) [inline docs/source/ko/model_doc/auto.md:19]:
```suggestion
많은 경우, 사용하려는 아키텍처는 `from_pretrained()` 메소드에서 제공하는 사전 학습된 모델의 이름이나 경로로부터 유추할 수 있습니다. AutoClasses는 이 작업을 위해 존재하며, 사전 학습된 모델 가중치/구성/단어사전에 대한 이름/경로를 제공하면 자동으로 관련 모델을 가져오도록 도와줍니다.
```

File: docs/source/ko/model_doc/auto.md [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+specific language governing permissions and limitations under the License.
+
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+
+# Auto 클래스[[auto-classes]]
+
+많은 경우, 사용하려는 아키텍처는 `from_pretrained()` 메소드에서 제공하는 사전 학습된 모델의 이름이나 경로로부터 유추할 수 있습니다. AutoClasses는 이 작업을 위해 존재하며, 사전 학습된 모델 가중치/구성/단어사전에 대한 이름/경로를 제공하면 자동으로 관련 모델을 가져오도록 도와줍니다.
```
Comment (boyunJang) [inline docs/source/ko/model_doc/auto.md:19]:
```suggestion
많은 경우, 사용하려는 아키텍처는 `from_pretrained()` 메소드에서 제공하는 사전 훈련된 모델의 이름이나 경로로부터 유추할 수 있습니다. AutoClasses는 이 작업을 위해 존재하며, 사전 학습된 모델 가중치/구성/단어사전에 대한 이름/경로를 제공하면 자동으로 관련 모델을 가져오도록 도와줍니다.
```

**Review by junejae — APPROVED** [review]
LGTM! 꼼꼼한 번역 감사드려요!

**Review by stevhliu — APPROVED** [review]
Thank you!

**Pull request merged**